vscode/src/completions/providers/anthropic.ts

```diff
@@ -169,30 +146,95 @@ export class AnthropicProvider extends Provider {
         tracer?.params(args)
 
         // Issue request
-        const responses = await batchCompletions(this.completionsClient, args, this.options.n, abortSignal)
+        const responses = await this.batchAndProcessCompletions(
+            this.completionsClient,
+            args,
+            this.options.n,
+            abortSignal
+        )
 
         // Post-process
-        const ret = responses.map(resp => {
-            const content = this.postProcess(resp.completion)
-
-            if (content === null) {
-                return []
-            }
-
-            return [
-                {
-                    prefix: this.options.prefix,
-                    content,
-                    stopReason: resp.stopReason,
-                },
-            ]
-        })
+        const ret = responses.map(resp => [
+            {
+                prefix: this.options.docContext.prefix,
+                content: resp.completion,
+                stopReason: resp.stopReason,
+            },
+        ])
 
         const completions = ret.flat()
         tracer?.result({ rawResponses: responses, completions })
 
         return completions
     }
+
+    private async batchAndProcessCompletions(
+        client: Pick<SourcegraphCompletionsClient, 'complete'>,
+        params: CompletionParameters,
+        n: number,
+        abortSignal: AbortSignal
+    ): Promise<CompletionResponse[]> {
+        const responses: Promise<CompletionResponse>[] = []
+        for (let i = 0; i < n; i++) {
+            responses.push(this.fetchAndProcessCompletions(client, params, abortSignal))
+        }
+        return Promise.all(responses)
+    }
+
+    private async fetchAndProcessCompletions(
+        client: Pick<SourcegraphCompletionsClient, 'complete'>,
+        params: CompletionParameters,
+        abortSignal: AbortSignal
+    ): Promise<CompletionResponse> {
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises, no-async-promise-executor
```

Comment:
I was wondering why do we need to disable ESLint rules here

```suggestion
        // The Async executor is required to return the completion early if a partial result from SSE can be used.
        // eslint-disable-next-line @typescript-eslint/no-misused-promises, no-async-promise-executor
```